src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -164,6 +167,17 @@ public string StorageConnectionString
         /// </summary>
         public TimeSpan? FunctionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether function invocations will timeout when
+        /// a <see cref="FunctionTimeout"/> is specified and a debugger is attached. False by default.
+        /// </summary>
+        public bool TimeoutWhileDebugging { get; set; }
+
+        /// <summary>
+        /// When true, an exception is thrown when a function timeout expires.
+        /// </summary>
+        public bool ThrowExceptionOnFunctionTimeout { get; set; }
```

Comment:
At the moment there's no grace period by default (but we also don't throw by default). In the Script repo I do this waiting in the ExceptionHandler -- and it's 2 seconds -- which is just an arbitrary starting point that I picked.

I could do something similar in the default ExceptionHandler. We could also have that grace period be configurable.

---
In reply to: [76662449](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r76662449) [](ancestors = 76662449)